pkg/commands/sso/root.go

```diff
@@ -108,17 +110,36 @@ func (c *RootCommand) Exec(in io.Reader, out io.Writer) error {
 	// For creating/updating a profile we set `prompt` because we want to ensure
 	// that another session (from a different profile) doesn't cause unexpected
 	// errors for the user flow. This forces a re-auth.
-	if c.InvokedFromProfileCreate || ForceReAuth {
+	switch {
+	case c.InvokedFromProfileCreate || ForceReAuth:
 		c.Globals.AuthServer.SetParam("prompt", "login select_account")
-	}
-	if c.InvokedFromProfileUpdate || c.InvokedFromProfileSwitch {
+	case c.InvokedFromProfileUpdate || c.InvokedFromProfileSwitch:
 		c.Globals.AuthServer.SetParam("prompt", "login")
 		if c.ProfileSwitchEmail != "" {
 			c.Globals.AuthServer.SetParam("login_hint", c.ProfileSwitchEmail)
 		}
 		if c.ProfileSwitchCustomerID != "" {
 			c.Globals.AuthServer.SetParam("account_hint", c.ProfileSwitchCustomerID)
 		}
+	default:
+		// Handle `fastly sso` being invoked directly.
+		p := profile.Get(profileName, c.Globals.Config.Profiles)
+		if p == nil {
+			err := fmt.Errorf(profile.DoesNotExist, profileName)
+			c.Globals.ErrLog.Add(err)
+			return err
+		}
+		c.Globals.AuthServer.SetParam("prompt", "login")
+		c.Globals.AuthServer.SetParam("login_hint", p.Email)
+		c.Globals.AuthServer.SetParam("account_hint", p.CustomerID)
+		// IMPORTANT: We must make the specified profile the default.
+		// If we don't, then the next command run will fail to check the token.
+		// Because the current profile will not be the active session.
+		// And we don't want to have to force the user to have re-auth again.
+		// Really, `fastly sso` should be a hidden command.
+		// And all users should use the `fastly profile ...` subcommands.
+		c.ProfileDefault = true
```

Comment:
Whilst I understand **_why_** we need to do this, it may be confusing to the user if we don't explicitly tell them we're doing this. Could we leave a message eitehr before or after the command saying that `Profile <foo> is now the default profile for all commands`.

Or we update the help text for this root command saying it is used to manage your default SSO profile?